### PR TITLE
Query programs by search keyword

### LINKED ISSUE #13: Query programs by searching name/description
**What is the expected feature or enhancement?**

A new search parameter has been added [here](https://runtime-us-east-dev.quantum-computing.ibm.com/openapi/#/Programs/list_programs). Users should be able to query by search keywords in the program name and description

### PR DESCRIPTION
**Pull request by kt474**
### Summary

Fixes #13 

### Details and comments

There will be merge conflicts with https://github.com/Qiskit/qiskit-ibm-runtime/pull/14

### CODE REVIEW

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]
@kt474 Use the same logic in my comment here https://github.com/Qiskit/qiskit-ibm-runtime/pull/14#issuecomment-984134337

File: qiskit_ibm_runtime/ibm_runtime_service.py [hunk 1]
```diff
@@ -171,6 +171,7 @@ def __init__(
         )
         self._backends: Dict[str, "ibm_backend.IBMBackend"] = {}
         self._api_client = None
+        self._programs_first_call = True
```
Comment (rathishcholarajan) [inline qiskit_ibm_runtime/ibm_runtime_service.py:174]:
```suggestion
```

File: qiskit_ibm_runtime/ibm_runtime_service.py [hunk 2]
```diff
@@ -908,6 +909,31 @@ def program(self, program_id: str, refresh: bool = False) -> RuntimeProgram:
 
         return self._programs[program_id]
 
+    def _retrieve_programs(self, search: str = "") -> None:
+        """Make an API call to fetch programs.
+
+        Args:
+            search: Search query for program name and description.
+        """
+        self._programs = {}
```
Comment (rathishcholarajan) [inline qiskit_ibm_runtime/ibm_runtime_service.py:918]: Same comment as here https://github.com/Qiskit/qiskit-ibm-runtime/pull/14#discussion_r763321347

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]

**Comment by kt474** [comment]
Closing and reopening https://github.com/Qiskit/qiskit-ibm-runtime/pull/78 to avoid switching between forks

**Pull request closed without merging**